.github/workflows/util/execute_and_retry.sh

```diff
@@ -3,28 +3,34 @@
 # This function is for retrying commands in the case they fail. It accepts three arguments
 # $1: Number of retries it will attempt
 # $2: Command to execute
-# $3: (Optional) Command for cleaning up resources if $2 fails
+# $3: Sleep time between run
+# $4: (Optional) Command for cleaning up resources if $2 fails
 execute_and_retry () {
   # Warning: The variables called in this function are not local and will be shared with the calling function.
   # Make sure that the variable names do not conflict
   execute_retry_counter=0
   max_execute_retry=$1
   command=$2
-  cleanup=$3
+  sleep_time=$3
+  cleanup=$4
+  echo "Initiating execute_and_retry.sh script for command $command"
   while [ $execute_retry_counter -lt $max_execute_retry ]; do
+   echo "Attempt Number $execute_retry_counter for execute_and_retry.sh"
    attempt_failed=0
    eval "$command" || attempt_failed=$?
 
    if [ $attempt_failed -ne 0 ]; then
+     echo "Command failed for execute_and_retry.sh, executing cleanup command for another attempt"
      eval "$cleanup"
      execute_retry_counter=$(($execute_retry_counter+1))
-     sleep 5
+     sleep $sleep_time
    else
+     echo "Command executed successfully for execute_and_retry.sh, exiting script"
      break
    fi
 
-   if [ $execute_retry_counter -eq $max_execute_retry ]; then
-     echo "Max retry reached, command failed to execute properly. Exiting code"
+   if [ "$execute_retry_counter" -eq "$max_execute_retry" ]; then
```

Comment:
nit: Can we use `-ge` here instead of `-eq` as well?

Reply:
Done